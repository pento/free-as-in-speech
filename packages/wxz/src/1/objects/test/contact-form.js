/**
 * External dependencies
 */
const FDBFactory = require( 'fake-indexeddb/lib/FDBFactory' );

/**
 * Internal dependencies
 */
const { WXRDriver } = require( '../../index.js' );

describe( 'Contact Form Object', () => {
	beforeEach( () => {
		window.indexedDB = new FDBFactory();
	} );

	test( 'All fields are written as expected', async () => {
		const wxr = new WXRDriver();
		await wxr.connect();

		wxr.addObject( 'contact-form', {
			email: 'gary@pento.net',
			fields: [
				{
					type: 'text',
					label: 'Name',
					required: true,
				},
				{
					type: 'text',
					label: 'Email',
					format: 'email',
					required: true,
				},
				{
					type: 'text',
					label: 'Phone',
					format: 'phone-number',
				},
				{
					type: 'text',
					label: 'Date',
					format: 'date',
					dateFormat: 'MM/DD/YYYY',
					required: true,
				},
				{
					type: 'select',
					label: 'Time',
					options: [
						{ text: '11:30 AM', value: '11:30 AM' },
						{ text: '12:00 PM', value: '12:00 PM' },
						{ text: '12:30 PM', value: '12:30 PM' },
						{ text: '1:00 PM', value: '1:00 PM' },
						{ text: '1:30 PM', value: '1:30 PM' },
						{ text: '2:00 PM', value: '2:00 PM' },
						{ text: '7:00 PM', value: '7:00 PM' },
						{ text: '7:30 PM', value: '7:30 PM' },
						{ text: '8:00 PM', value: '8:00 PM' },
						{ text: '8:30 PM', value: '8:30 PM' },
						{ text: '9:00 PM', value: '9:00 PM' },
						{ text: '9:30 PM', value: '9:30 PM' },
					],
					required: true,
				},
				{
					type: 'textarea',
					label: 'Special Request',
					placeholder: '',
				},
				{ type: 'submit', label: 'Submit' },
			],
		} );

		const xml = await wxr.export();

		expect( xml ).toMatchSnapshot();
	} );
} );
