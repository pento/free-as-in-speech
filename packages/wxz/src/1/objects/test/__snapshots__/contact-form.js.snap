// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Form Object All fields are written as expected 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" ?>

<!-- This is a WordPress eXtended RSS file generated by the \\"Free (as in Speech)\\" browser extension, as an export of your site. -->
<!-- It contains information about your site's posts, pages, comments, categories, and other content. -->
<!-- You may use this file to transfer that content from one site to another. -->
<!-- This file is not intended to serve as a complete backup of your site. -->

<!-- To import this information into a WordPress site follow these steps: -->
<!-- 1. Log in to that site as an administrator. -->
<!-- 2. Go to Tools: Import in the WordPress admin panel. -->
<!-- 3. Install the \\"WordPress\\" importer from the list. -->
<!-- 4. Activate & Run Importer. -->
<!-- 5. Upload this file using the form provided on that page. -->
<!-- 6. You will first be asked to map the authors in this export file to users -->
<!--    on the site. For each author, you may choose to map to an -->
<!--    existing user on the site or to create a new user. -->
<!-- 7. WordPress will then import each of the posts, pages, comments, categories, etc. -->
<!--    contained in this file into your site. -->

<rss version=\\"2.0\\"
	xmlns:excerpt=\\"http://wordpress.org/export/1.3/excerpt/\\"
	xmlns:content=\\"http://purl.org/rss/1.0/modules/content/\\"
	xmlns:wfw=\\"http://wellformedweb.org/CommentAPI/\\"
	xmlns:dc=\\"http://purl.org/dc/elements/1.1/\\"
	xmlns:wp=\\"http://wordpress.org/export/1.3/\\"
>
<channel>

	<wp:object xmlns=\\"http://wordpress.org/export/objects/contact-form/1.0/\\" id=\\"1\\">
		<email>gary@pento.net</email>
		<field type=\\"text\\" label=\\"Name\\" required/>
		<field type=\\"text\\" label=\\"Email\\" format=\\"email\\" required/>
		<field type=\\"text\\" label=\\"Phone\\" format=\\"phone-number\\"/>
		<field type=\\"text\\" label=\\"Date\\" format=\\"date\\" dateFormat=\\"MM/DD/YYYY\\" required/>
		<field type=\\"select\\" label=\\"Time\\" required>
			<option value=\\"11:30 AM\\">11:30 AM</option>
			<option value=\\"12:00 PM\\">12:00 PM</option>
			<option value=\\"12:30 PM\\">12:30 PM</option>
			<option value=\\"1:00 PM\\">1:00 PM</option>
			<option value=\\"1:30 PM\\">1:30 PM</option>
			<option value=\\"2:00 PM\\">2:00 PM</option>
			<option value=\\"7:00 PM\\">7:00 PM</option>
			<option value=\\"7:30 PM\\">7:30 PM</option>
			<option value=\\"8:00 PM\\">8:00 PM</option>
			<option value=\\"8:30 PM\\">8:30 PM</option>
			<option value=\\"9:00 PM\\">9:00 PM</option>
			<option value=\\"9:30 PM\\">9:30 PM</option>
		</field>
		<field type=\\"textarea\\" label=\\"Special Request\\" placeholder=\\"\\"/>
		<field type=\\"submit\\" label=\\"Submit\\"/>
	</wp:object>
</channel>
</rss>"
`;
